### ISSUE TITLE
Testing Improvments

### ISSUE THREAD

**Issue by tyler-smith** [OpenBazaar/openbazaar-go#1151]
## Goal

Tests should be fast, thorough, and parallizable.
Tests should test all states and failure paths.
Tests should  thoroughly exercise local services. (e.g. local IPFS and SQLite)
Tests should not interact with outside services. (e.g. ticker and fee APIs)

## Acceptance Criteria
- [ ] Each package has at least 60% coverage
- [ ] Overall openbazaar-go has at least 70% coverage
- [ ] All packages can be tested in parallel (#1145)
- [ ] Each package parallizes tests as much as possible
- [ ] Error paths should be tested within a reasonable level of effort
- [ ] All interactions with networked services mocked (i.e. offline tests)
- [ ] Packages with tests taking longer than 2 seconds should be checked for potential optimizations

## Stretch Goals
- [ ] Tests work with `-race` flag
- [ ] All tests complete in <= 60s

## Beginning State

Runtime: 139.64s (130-140s on average)
Overall: 34.673%
Highest: 73.9%
Lowest: None

```
ok  	github.com/OpenBazaar/openbazaar-go/api	3.696s	coverage: 24.4% of statements
?   	github.com/OpenBazaar/openbazaar-go/bitcoin	[no test files]
?   	github.com/OpenBazaar/openbazaar-go/bitcoin/listeners	[no test files]
ok  	github.com/OpenBazaar/openbazaar-go/core	0.389s	coverage: 6.0% of statements
ok  	github.com/OpenBazaar/openbazaar-go/ipfs	0.132s	coverage: 11.8% of statements
?   	github.com/OpenBazaar/openbazaar-go/mobile	[no test files]
ok  	github.com/OpenBazaar/openbazaar-go/net	6.167s	coverage: 50.3% of statements
?   	github.com/OpenBazaar/openbazaar-go/net/service	[no test files]
?   	github.com/OpenBazaar/openbazaar-go/net/repointer	[no test files]
?   	github.com/OpenBazaar/openbazaar-go/net/retriever	[no test files]
ok  	github.com/OpenBazaar/openbazaar-go/repo	0.195s	coverage: 57.4% of statements
ok  	github.com/OpenBazaar/openbazaar-go/repo/db	11.280s	coverage: 82.2% of statements
ok  	github.com/OpenBazaar/openbazaar-go/repo/migrations	12.422s	coverage: 72.8% of statements
ok  	github.com/OpenBazaar/openbazaar-go/schema	0.250s	coverage: 61.3% of statements
?   	github.com/OpenBazaar/openbazaar-go/storage	[no test files]
?   	github.com/OpenBazaar/openbazaar-go/storage/dropbox	[no test files]
ok  	github.com/OpenBazaar/openbazaar-go/storage/selfhosted	0.073s	coverage: 73.9% of statements
```
